apps/backend/src/main/java/tabletop/gather/backend/game/GameResource.java

```diff
@@ -15,41 +15,75 @@
-        final UUID createdId = gameService.create(gameDto);
-        return new ResponseEntity<>(createdId, HttpStatus.CREATED);
-    }
-
-    @PutMapping("/{id}")
-    public ResponseEntity<UUID> updateGame(@PathVariable(name = "id") final UUID id,
-            @RequestBody @Valid final GameDto gameDto) {
-        gameService.update(id, gameDto);
-        return ResponseEntity.ok(id);
-    }
-
-    @DeleteMapping("/{id}")
-    @ApiResponse(responseCode = "204")
-    public ResponseEntity<Void> deleteGame(@PathVariable(name = "id") final UUID id) {
-        gameService.delete(id);
-        return ResponseEntity.noContent().build();
-    }
+  private final GameService gameService;
 
+  private final JwtService jwtService;
+
+  public GameResource(final GameService gameService, final JwtService jwtService) {
+    this.gameService = gameService;
+    this.jwtService = jwtService;
+  }
+
+  /**
+   * Get all games by containing name
+   *
+   * @param name the name of the game
+   * @return all games filtered by containing name
+   */
+  @GetMapping
+  public ResponseEntity<List<GameDto>> getAllGames(final String name) {
+    return ResponseEntity.ok(gameService.findByUserId(name));
+  }
+
+  /**
+   * Get all games by user
+   *
+   * @param userId the id of the user
+   * @return all games filtered by user
+   */
+  @GetMapping("/user/{userId}")
+  public ResponseEntity<List<GameDto>> getGamesByUser(@PathVariable(name = "userId") final UUID userId) {
+    return ResponseEntity.ok(gameService.findByUserId(userId));
+  }
+
+  /**
+   * Get a game by id
+   *
+   * @param id the id of the game
+   * @return the game
+   */
+  @GetMapping("/{id}")
+  public ResponseEntity<GameDto> getGame(@PathVariable(name = "id") final UUID id) {
+    return ResponseEntity.ok(gameService.get(id));
+  }
+
+  /**
+   * Add a game to collection of authenticated user
+   *
+   * @param id the game to add
+   * @return 201 if added
+   */
+  @PostMapping("/{id}/add")
+  @ApiResponse(responseCode = "201")
+  public ResponseEntity<Void> addGameToUser(@RequestHeader("Authorization") final String token,
+                                            @PathVariable(name = "id") final UUID id) {
+    UUID userId = jwtService.getUserByToken(token).getId();
+    gameService.addUser(id, userId);
+    return ResponseEntity.status(HttpStatus.CREATED).build();
+  }
+
+  /**
+   * Remove a game from collection of authenticated user
+   *
+   * @param id the game to remove
+   * @return 204 if removed
+   */
+  @DeleteMapping("/{id}/remove")
+  @ApiResponse(responseCode = "204")
+  public ResponseEntity<Void> removeGameFromUser(@RequestHeader("Authorization") final String token,
+                                                 @PathVariable(name = "id") final UUID id) {
+    UUID userId = jwtService.getUserByToken(token).getId();
+    gameService.removeUser(id, userId);
+    return ResponseEntity.noContent().build();
+  }
```

Comment:
Da bruchemer no ä `getMyGames` Endpoint für z erstelle vomne Plan. Eine wo über z Jwt geit wie bide Plans.

Reply:
Würds nid meh Sinn mache we mir d userId im localStorage ablege bim login und frages übere Endpoint mit dr userId ab?

Reply:
Schad füre Ufwand. Wär ja viu eifacher im Backend. Chame eifach bide Plans abluege
```java
  /**
   * Get all my plans
   * 
   * @return all plans
   */
  @GetMapping("/me")
  public ResponseEntity<List<OverviewPlanDto>> getAllPlans(@RequestHeader("Authorization") final String token) {
    UUID userId = jwtService.getUserByToken(token).getId();
    return ResponseEntity.ok(planService.findAll(userId));
  }
```